api/lib/gateways/Academy-Benefits/FetchRecord.js

```diff
@@ -1,28 +1,49 @@
 const path = require('path');
-const {
-  checkString,
-  nameCase,
-  formatAddress,
-  formatRecordDate,
-  upperCase,
-  loadSQL
-} = require('../../Utils');
-const { Systems, IncomeFrequency } = require('../../Constants');
-const {
-  fetchCustomerSQL,
-  fetchCustomerBenefitsSQL,
-  fetchCustomerHouseholdSQL
-} = loadSQL(path.join(__dirname, 'sql'));
+const { nameCase, formatRecordDate, loadSQL } = require('../../Utils');
+const { fetchCustomerBenefitsSQL, fetchCustomerHouseholdSQL } = loadSQL(
+  path.join(__dirname, 'sql')
+);
+const { IncomeFrequency } = require('../../Constants');
 
-async function fetchCustomer(id, db) {
+async function fetchCustomerDb(claim_id, person_ref, fetchDB, logger) {
+  try {
+    return await fetchDB.execute(claim_id, person_ref);
+  } catch (err) {
+    logger.error(`Error fetching customers in Academy-Benefits: ${err}`, err);
+  }
+}
+
+async function fetchCustomerAPI(claim_id, person_ref, fetchAPI, logger) {
+  try {
+    return await fetchAPI.execute(claim_id, person_ref);
+  } catch (err) {
+    logger.error(
+      `Error fetching customers in Academy-Benefits API: ${err}`,
+      err
+    );
+  }
+}
+
+async function fetchCustomer(id, fetchDB, fetchAPI, logger) {
   const [claim_id, person_ref] = id.split('/');
+  const customer = await fetchCustomerDb(claim_id, person_ref, fetchDB, logger);
+  const customerAPI = await fetchCustomerAPI(
+    claim_id,
+    person_ref,
+    fetchAPI,
+    logger
+  );
 
-  return (
-    await db.request(fetchCustomerSQL, [
-      { id: 'claim_id', type: 'NVarChar', value: claim_id.slice(0, 7) },
-      { id: 'person_ref', type: 'Int', value: person_ref }
-    ])
-  )[0];
+  if (customer == customerAPI) {
```

Comment:
Are we comparing two different objects here?

We might need to compare each property individually (and check that neither are missing any properties)?

The associated test below seems to pass in two references to the same object for this comparison, possibly?

Reply:
(Sorry, [this](https://github.com/LBHackney-IT/housing-needs-single-view-api/pull/102/files#diff-69f62a155bbf9ddc1bbbc041213794a7R156) is the associated test I mentioned.)

Reply:
So, we are comparing the fetchCustomerDB and fetchcustomerAPI to check if they hold the same values

Are you suggesting to compare individuals (e.g. compare person_ref from fetchDB and person_ref from fetchAPI)